client/src/backoffice/InboxPage.tsx

```diff
@@ -1,202 +1,208 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+// libraries
 import classNames from 'classnames';
 import { Formik, Form, Field, ErrorMessage, FieldProps, FormikActions } from 'formik';
-import { InboxFormSchema } from '../lib/schemas';
+import { useToasts } from 'react-toast-notifications';
 
 /* Helpers */
 import { getEvents, PoapEvent, sendNotification } from '../api';
 
 /* Components */
 import { SubmitButton } from '../components/SubmitButton';
+import FilterSelect from '../components/FilterSelect';
 
-/* Typings */
-import { Value } from '../types';
+// schema
+import { InboxFormSchema } from '../lib/schemas';
 
-interface IInboxState {
-  events: PoapEvent[];
-  initialValues: IInboxFormValues;
-}
+/* Typings */
+import { Name, NotificationType, RecipientType } from '../types';
 
-interface IInboxFormValues {
+type IInboxFormValues = {
   title: string;
   description: string;
-  recipientFilter: Exclude<Value, 'inbox' | 'push'>;
-  selectedEventId: number | null;
-  notificationType: Exclude<Value, 'everyone' | 'event'>;
-}
-
-export class InboxPage extends React.Component<{}, IInboxState> {
-  state: IInboxState = {
-    events: [],
-    initialValues: {
-      title: '',
-      description: '',
-      recipientFilter: 'everyone',
-      selectedEventId: null,
-      notificationType: 'inbox',
-    },
-  };
-
-  async componentDidMount() {
-    const events = await getEvents();
+  recipientFilter: RecipientType;
+  notificationType: NotificationType;
+  selectedEvent: number | null;
+};
 
-    this.setState({ events });
-  }
+export const InboxPage: React.FC = () => {
+  const [events, setEvents] = useState<PoapEvent[]>([]);
+  const [notificationType, setNotificationType] = useState<string>('');
+  const [recipientFilter, setRecipientFilter] = useState<string>('');
+  const [selectedEvent, setSelectedEvent] = useState<number | null>();
+
+  const { addToast } = useToasts();
+
+  useEffect(() => {
+    const async = async () => {
+      const events = await getEvents();
+      setEvents(events);
+    };
+    async();
+  }, []);
```

Comment:
There's not too much benefits here on using `async/await` unless we will do some data parsing in a near future.
Prefer a simple Promise call:

```typescript
useEffect(() => {
  getEvents().then(setEvents);
}, []);
```